src/Http/WebUtilities/test/FormPipeReaderTests.cs

```diff
@@ -284,6 +284,32 @@ public void TryParseFormValues_MultiSegmentSplitAcrossSegmentsThatNeedDecodingWo
             Assert.Equal("wow", dict["\"%-.<>\\^_`{|}"]);
         }
 
+        [Fact]
+        public void TryParseFormValues_MultiSegmentExceedKeyLengthThrows()
+        {
+            var readOnlySequence = ReadOnlySequenceFactory.CreateSegments(Encoding.UTF8.GetBytes("foo=bar&baz=bo"), Encoding.UTF8.GetBytes("o&t="));
+
+            KeyValueAccumulator accumulator = default;
+
+            var formReader = new FormPipeReader(null);
+            formReader.KeyLengthLimit = 2;
+
+            Assert.Throws<InvalidDataException>(() => formReader.ParseFormValues(ref readOnlySequence, ref accumulator, isFinalBlock: true));
+        }
+
+        [Fact]
+        public void TryParseFormValues_MultiSegmentExceedValueLengthThrows()
+        {
+            var readOnlySequence = ReadOnlySequenceFactory.CreateSegments(Encoding.UTF8.GetBytes("foo=bar&baz=bo"), Encoding.UTF8.GetBytes("o&t="));
```

Comment:
`bar` needs to be shorter if you're trying to test `boo` spanning the sequence gap.